oktoberfest/plotting/plotting.py

```diff
@@ -452,11 +461,134 @@ def plot_sa_distribution(prosit_df: pd.DataFrame, target_df: pd.DataFrame, decoy
     plt.close()
 
 
-def plot_all(data_dir: Path):
+def plot_mirror_spectrum(
+    spec_pred: Spectra,
+    mzml: pd.DataFrame,
+    raw_file: str,
+    scan_number: int,
+    config: Config,
+    prosit_df: pd.DataFrame,
+    target_df: pd.DataFrame,
+    decoy_df: pd.DataFrame,
+    pdf: PdfPages,
+):
+    """
+    Generate a mirror plot comparing an experimental and predicted MS/MS spectrum.
+
+    :param spec_pred: Spectra object containing predicted MS/MS spectra
+    :param mzml: ThermoRaw object containing experimental MS/MS spectra from an mzML file
+    :param raw_file: The name of the raw file being processed
+    :param scan_number: The scan number of the spectrum to be plotted
+    :param config: the configuration object
+    :param prosit_df: mokapot / percolator input tab for rescoring with peptide property prediction
+    :param target_df: mokapot / percolator target output for rescoring with peptide property prediction on the psm level
+    :param decoy_df: mokapot / percolator decoy output for rescoring with peptide property prediction on the psm level
+    :param pdf: PDF file object for saving mirror plots
+
+    :raises ValueError: If the mass analyzer type is unknown.
+    """
+    score_col, _, _, spec_col = _check_columns(target_df)
+    target_df["target"] = True
+    decoy_df["target"] = False
+    concat_target_decoy = pd.concat([target_df, decoy_df])
+    if spec_col == "PSMId":
+        concat_target_decoy["ScanNr"] = concat_target_decoy["PSMId"].str.split("-").str[1].astype(int)
+    filtered_obs = spec_pred.obs[spec_pred.obs["SCAN_NUMBER"] == scan_number]
+    if filtered_obs.empty:
+        print(f"Warning: Scan number {scan_number} for {raw_file} not found in the prediction file.")
+        return
+    obs = filtered_obs.iloc[0]
+
+    mod_sequence = obs["MODIFIED_SEQUENCE"].replace("[]-", "").replace("-[]", "")
+    charge = obs["PRECURSOR_CHARGE"]
+    mass = obs["MASS"]
+    mass_analyzer = obs["MASS_ANALYZER"]
+    fragm = obs["FRAGMENTATION"]
+    raw_file = obs["RAW_FILE"]
+    rt = obs["RETENTION_TIME"].round(2)
+    ce = obs["COLLISION_ENERGY"]
+    model = config.models["intensity"]
+    ion_types = config.ion_types
+    abs_rt_diff = prosit_df[(prosit_df["ScanNr"] == scan_number) & (prosit_df["filename"] == raw_file)][
+        "abs_rt_diff"
+    ].iloc[0]
+    sa = prosit_df[(prosit_df["ScanNr"] == scan_number) & (prosit_df["filename"] == raw_file)]["spectral_angle"].iloc[0]
+    ce = prosit_df[(prosit_df["ScanNr"] == scan_number) & (prosit_df["filename"] == raw_file)][
+        "collision_energy_aligned"
+    ].iloc[0]
+    score = concat_target_decoy[
+        (concat_target_decoy["ScanNr"] == scan_number) & (concat_target_decoy["filename"] == raw_file)
+    ][score_col].iloc[0]
+
+    # Set tolerance based on mass analyzer
```

Comment:
check the function in spectrum_fundamentals for this, you should provide the option to have non-defaults read from the config file